### PR TITLE
AF-2762: UI exception when creating a new Page

### PR DESCRIPTION
**Pull request by jesuino**
FIX for https://issues.redhat.com/browse/AF-2762

### CODE REVIEW

**Comment by jesuino** [comment]
jenkins execute fdb

File: dashbuilder/dashbuilder-backend/dashbuilder-external-backend/src/main/java/org/dashbuilder/external/impl/ComponentServiceImpl.java [hunk 1]
```diff
@@ -55,11 +56,13 @@
     
     @Override
     public List<ExternalComponent> listAllComponents() {
-        List<ExternalComponent> allComponents = loader.loadExternal();
+        List<ExternalComponent> allComponents = new ArrayList<>();
+        List<ExternalComponent> external = loader.loadExternal();
```
Comment (caponetto) [inline dashbuilder/dashbuilder-backend/dashbuilder-external-backend/src/main/java/org/dashbuilder/external/impl/ComponentServiceImpl.java:60]: The list `external` seems to be useless here. Am I missing something?

**Review by caponetto — APPROVED** [review]

File: dashbuilder/dashbuilder-backend/dashbuilder-external-backend/src/main/java/org/dashbuilder/external/impl/ComponentServiceImpl.java (hunk 1 above)
Comment (jesuino) [reply dashbuilder/dashbuilder-backend/dashbuilder-external-backend/src/main/java/org/dashbuilder/external/impl/ComponentServiceImpl.java:60]: You are right, I forgot this line, added in my latest commit and enabled the test that checks it (strangely the test didn't fail - the reason was it was not enabled!!)

Comment (caponetto) [reply dashbuilder/dashbuilder-backend/dashbuilder-external-backend/src/main/java/org/dashbuilder/external/impl/ComponentServiceImpl.java:60]: Great! Now we are safe :)

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
Hello @sthundat and @barboras7 do you know who could verify this PR? The issue was an error when entering the pages perspective.

Thanks!

**Review by barboras7 — APPROVED** [review]
Looks good to me, thanks @jesuino!

**Comment by jesuino** [comment]
Failed tests are unrelated

```
org.kie.server.springboot.samples.KieServerTest.testProcessStartAndAbortUsingAlias
org.kie.server.springboot.samples.KieServerTest.testProcessStartAndAbort
org.kie.server.springboot.samples.KieServerTest.testProcessStartAndWorkOnUserTask
org.kie.hacep.LocalStorageStreamingKieSessionTest.getCommandsTest
```

**Review by ederign — APPROVED** [review]

**Pull request merged**